src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -1270,5 +1269,30 @@ protected virtual IEnumerable<SelectListItem> GetEnumSelectList(ModelMetadata me
 
             return selectList;
         }
+
+        // This method exists only because we cannot add IHtmlGenerator.CreateSanitizedId() until 2.0.
+        private string CreateSanitizedId(string fullName)
+        {
+            // First, cover possibility user has subclassed DefaultHtmlGenerator and overridden its CreateSanitizedId()
+            // implementation.
+            var defaultGenerator = _htmlGenerator as DefaultHtmlGenerator;
+            if (defaultGenerator != null)
+            {
+                return defaultGenerator.CreateSanitizedId(ViewContext, fullName);
+            }
+
+            // Fall back to duplicating DefaultHtmlGenerator.CreateSanitizedId() when user has replaced the
+            // IHtmlGenerator implementation with something unrelated.
+            var lastSanitizedId = ViewContext.LastSanitizedId;
```

Comment:
Because DHG is a singleton anyhow. Mixing that with `internal static` crud is just cruddy.